### PR TITLE
Divide build-binaries job to multiple jobs per platform

### PR DESCRIPTION
**Pull request by Ashmita152**
Signed-off-by: Ashmita Bohara <ashmita.bohara152@gmail.com>

## Which problem is this PR solving?
Right now build-binaries is taking around 30 minutes to complete. This should do this in parallel.

## Short description of the changes
Moving each platform binaries build into separate job.

### CODE REVIEW

**Review by yurishkuro — DISMISSED** [review]
Sweet!

 * Build binaries / binaries linux (pull_request) Successful in **9m**

File: .github/workflows/ci-build-binaries.yml [hunk 1]
```diff
@@ -9,6 +9,22 @@ on:
...
+          task: build-binaries-windows
+        - name: osx
+          task: build-binaries-darwin
+        - name: system/390
+          task: build-binaries-s390x
+        - name: arm
+          task: build-binaries-arm64
+        - name: powerpc
+          task: build-binaries-ppc64le
+    name: binaries ${{ matrix.platform.name }}
```
Comment (yurishkuro) [inline .github/workflows/ci-build-binaries.yml:27]:
```suggestion
    name: build binaries for ${{ matrix.platform.name }}
```

**Comment by yurishkuro** [comment]
One unfortunate side effect is that it explodes the number of jobs we need to configure as `required` for the branch to pass:
image

File: .github/workflows/ci-build-binaries.yml (hunk 1 above)
Comment (yurishkuro) [reply .github/workflows/ci-build-binaries.yml]: nvm lets keep as is

**Comment by Ashmita152** [comment]
> One unfortunate side effect is that it explodes the number of jobs we need to configure as required for the branch to pass:
I see. Hopefully we don't hit any limit imposed by github.

Yuri, I have one question: I remember you mentioning that jaegertracing org should get more credits as compared to personal open-source project. But from my experience I haven't seen any difference in the build times for a job in my jaeger's fork vs same build in jaeger.

**Comment by yurishkuro** [comment]
> I remember you mentioning that jaegertracing org should get more credits as compared to personal open-source project.

I am not sure about that. This [page][1] shows 2000 minutes for both personal and orgs. I am not sure if this is enforced yet, because I would've thought we easily exceed 2000 minutes.

It would be nice if GHA were able to automatically cancel running jobs for stale commits, we're burning through those a lot.

[1]: https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions

**Comment by Ashmita152** [comment]
> It would be nice if GHA were able to automatically cancel running jobs for stale commits, we're burning through those a lot.

I will look into it. Thinking if we can do something like this:

- Create a custom GHA to kill all GHA.
- Add the above custom action as a step if any of the task fails in the jobs which are marked as `required`.

**Comment by yurishkuro** [comment]
I am not sure about the 2nd point. For the first one, there are already GHAs that kill other runs, but when I looked at few of them none were particularly great (may have changed since then)

**Pull request merged**